wavelength-plugins/lottery.js

```diff
@@ -0,0 +1,121 @@
+				Economy.readMoney(user.userid, money => {
+					Economy.logTransaction(user.userid + " entered a Lottery drawing for " + this.costToJoin + " " + currencyPlural + ".");
+				});
+			});
+		});
+		this.playerCount++;
+	}
+
+	//TODO: Add a check that the user must be already in the Lottery drawing before able to leave
+	leaveLottery(user) {
+		user.sendTo(this.room, '|html|' + WL.nameColor(user.name, true) + ' has left the game!');
+		Economy.writeMoney(user.userid, this.costToJoin);
+		Economy.logTransaction(user.userid + " has left the Lottery drawing, and has been refunded their " + this.costToJoin + " " + currencyPlural + ".");
+		this.playerCount--;
+	}
+
+	end(user) {
+		this.room.add(`|uhtmlchange|lottery-${this.lottoNumber}|<div class="infobox">This Lottery Drawing has ended.</div>`, true);
+		clearTimeout(this.timer);
+		delete this.room.lottery;
+	}
+}
+
+exports.commands = {
+	lotto: "lottery",
+	lottery: {
+		create: "new",
+		make: "new",
+		new: function (target, room, user) {
+			if (room.lottery) return this.sendReply("A join-able Lottery drawing is already active.");
+			if (!this.can('broadcast', null, room)) return false;
+			if (!this.room.isOfficial) return this.sendReply('Lottery drawings can only be created in Official Chatrooms.');
+			this.privateModCommand(`(A new Lottery drawing has been created.)`);
+			room.lottery = new Lottery(room, user);
+		},
+		j: "join",
+		join: function (target, room, user) {
+			if (!this.canTalk()) return this.sendReply("You must be able to talk to join a Lottery drawing.");
+			if (!this.state === "signups") return this.sendReply('You cannot join a Lottery drawing after it has started.');
+			if (!user.registered) return this.sendReply("To join the Lottery, you must be on a registered account.");
+			if (!room.lottery) return this.sendReply("There is no join-able Lottery drawing going on right now.");
+			room.lottery.joinLottery(user, this);
+		},
+		part: "leave",
+		l: "leave",
+		leave: function (target, room, user) {
+			if (!room.lottery) return this.sendReply("There is no active Lottery drawing in this room.");
+			room.lottery.leaveLottery(user, this);
+		},
+		forcestart: "start",
+		begin: "start",
+		start: function (target, room, user) {
+			if (!this.can('broadcast', null, room)) return;
+			if (!room.lottery) return this.sendReply("There is not any Lottery drawing available to be started.");
+			if (room.lottery.playerCount < 1) return this.sendReply("You can't start a Lottery drawing without at least one user joining.");
+			this.privateModCommand(`(A new Lottery drawing has been started early.)`);
+			room.lottery.drawWinner(user, this);
+		},
+		cancel: "end",
+		end: function (target, room, user) {
+			if (!this.can('broadcast', null, room)) return;
+			if (!room.lottery) return this.sendReply("There is no Lottery drawing going on right now.");
+			this.privateModCommand(`(The Lottery drawing was forcefully ended.)`);
+			room.lottery.end(user, this);
```

Comment:
Why are you passing `this` when it isnt used in the method?

user also isnt used in the method, remove it here if you remove it there.